src/component/Options.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { createRef, useEffect, useRef, useState } from 'react';
+
+import { PROFILE_URL } from '../auth';
+import { getStorageItem, setStorageItem } from '../storage';
+import Box from '@mui/material/Box';
+import { TextField, Button, Link, Typography, Snackbar, Alert } from '@mui/material';
+import SettingsIcon from '@mui/icons-material/Settings';
+import OpenInNewIcon from '@mui/icons-material/OpenInNew';
+import LoginIcon from '@mui/icons-material/Login';
+import SaveAltIcon from '@mui/icons-material/SaveAlt';
+import CloseIcon from '@mui/icons-material/Close';
+import ChecklistRtlIcon from '@mui/icons-material/ChecklistRtl';
+import { v4 as uuidv4 } from 'uuid';
+
+const EditableList = () => {
+  const [text, setText] = useState('');
+  const [open, setOpen] = useState(false);
+  const [severity, setSeverity] = useState<'success' | 'error'>('success');
+  const [message, setMessage] = useState('');
+
+  useEffect(() => {
+    (async () => {
+      const whitelist = await getStorageItem('whitelist');
+      setText(whitelist.join('\n'));
+    })();
+  }, []);
+
+  return (
+    <>
+      <Typography variant="h6">
+        Whitelist{' '}
+        <ChecklistRtlIcon
+          sx={{
+            verticalAlign: 'bottom',
+          }}
+          fontSize="large"
+        />
+      </Typography>
+      <Typography variant="body2">
+        Domains to allow auto-completion, double-click to edit, use regex format
+      </Typography>
+
+      <TextField
+        label="Domain Regexes"
+        variant="standard"
+        fullWidth
+        value={text}
+        multiline
+        rows={10}
+        sx={{
+          '& .MuiInputBase-root': {
+            fontSize: '0.8rem',
+            fontFamily: 'monospace',
+          },
+        }}
+        onChange={(e) => setText(e.target.value)}
+      ></TextField>
+
+      <Button
+        variant="text"
+        onClick={() => {
+          const lst = text.split('\n').map((x) => x.trim());
+          for (const rule of lst) {
+            try {
+              new RegExp(rule);
+            } catch (e) {
+              setSeverity('error');
+              setMessage(`Invalid regex: ${rule}`);
+              setOpen(true);
+              return;
+            }
+          }
+          setStorageItem('whitelist', lst);
+          setSeverity('success');
+          setMessage('Saved successfully');
+          setOpen(true);
+        }}
+        sx={{
+          float: 'right',
+          textTransform: 'none',
+        }}
+      >
+        Save <SaveAltIcon />
```

Comment:
It's ambiguous from the UI if the Save button will also save the token field or not.